R/update-input.R

```diff
@@ -667,10 +666,23 @@ selectizeJSON <- function(data, req) {
   # convert a single vector to a data frame so it returns {label: , value: }
   # later in JSON; other objects return arbitrary JSON {x: , y: , foo: , ...}
   data <- if (is.atomic(data)) {
-    data.frame(label = names(choicesWithNames(data)), value = data,
+    lab = if(is.null(names(data))) {
+      as.character(data)
+    } else {
+      lab = names(data)
```

Comment:
The nested assignment of `lab` is confusing. It would be better to do `lab <-` in each of the paths of the if-else.